### PR TITLE
task-07 routing

### PR DESCRIPTION
**Pull request by ShevtsovAnton**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,10 +1,28 @@
...
+    canActivate: [AuthGuard]
+  },
+  {
+    path: 'courses/:id',
+    component: AddEditCoursePageComponent,
+    pathMatch: 'full',
+    canActivate: [AuthGuard]
+  },
+  { path: 'courses', component: CoursesPageComponent, pathMatch: 'full', canActivate: [AuthGuard] },
+  { path: '', redirectTo: '/courses', pathMatch: 'full' },
```
Comment (vnovikov-clgx) [inline src/app/app-routing.module.ts]: Minor: In the feature try to set routes from root to end routes

File: src/app/auth.guard.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot
+  ): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    return this.authorizationService.isAuthenticated()
+      .pipe(
+        map(isAuthenticated => {
+          if (isAuthenticated) {
+            return true;
+          } else {
+            this.router.navigate(['/login']);
```
Comment (vnovikov-clgx) [inline src/app/auth.guard.ts]: in this case you should return `UrlTree`

Comment (vnovikov-clgx) [reply src/app/auth.guard.ts]: you can make smth like this:
 ```
isAuthenticated: addType => !!isAuthenticated || this.router.createUrlTree([/login])
```

Comment (vnovikov-clgx) [reply src/app/auth.guard.ts]: Minor: pls create enum with all routes, it's make easer feature cahnges

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (ShevtsovAnton) [reply src/app/app-routing.module.ts]: changed the order of routes

File: src/app/auth.guard.ts (hunk 2 above)
Comment (ShevtsovAnton) [reply src/app/auth.guard.ts]: 1. Changed to return UlrTree
2. created enums with routes

**Pull request merged**